src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3058,26 +3060,32 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                 int addedCount = 0;
                 foreach (ParameterSymbol param in recordParameters)
                 {
-                    var property = new SynthesizedRecordPropertySymbol(this, param, diagnostics);
-                    _ = memberSignatures.TryGetValue(property, out var existingMember);
-                    existingMember ??= getInheritedMember(property, this);
+                    bool isInherited = false;
+                    var syntax = param.GetNonNullSyntaxNode();
+                    var property = new SynthesizedRecordPropertySymbol(this, syntax, param, isOverride: false, diagnostics);
+                    if (!memberSignatures.TryGetValue(property, out var existingMember))
+                    {
+                        Debug.Assert(property.OverriddenOrHiddenMembers.OverriddenMembers.Length == 0); // property is not virtual and should not have overrides
+                        existingMember = property.OverriddenOrHiddenMembers.HiddenMembers.FirstOrDefault();
```

Comment:
>property.OverriddenOrHiddenMembers.HiddenMembers.FirstOrDefault(); [](start = 41, length = 66)

Do we care if the hidden member is not a property?

Reply:
> Do we care if the hidden member is not a property?

Not currently we don't. We report an error if the property hides a distinct member. (That is existing behavior.)

---
In reply to: [446540627](https://github.com/dotnet/roslyn/pull/45336#discussion_r446540627) [](ancestors = 446540627)